src/remoteClient/clientContextImpl.cpp

```diff
@@ -4351,6 +4355,38 @@ class InternalClientContextImpl :
         BeaconHandler::shared_pointer handler;
         if (it == m_beaconHandlers.end())
         {
+            TimeStamp now;
+            now.getCurrent();
+
+            double oldestAge = -1;
+            AddressBeaconHandlerMap::iterator oldestBeacon = m_beaconHandlers.end();
+
+            /* Before creating a new beacon, cleanup any old ones */
+            for (AddressBeaconHandlerMap::iterator it = m_beaconHandlers.begin(); it != m_beaconHandlers.end();)
```

Comment:
This is a linear iteration of the list of tracked servers each time a beacon is received.  If the number of servers grows large, or one gets stuck in a restart loop, this could potentially take up a lot of CPU time.

`Timer InternalClientContextImpl::m_timer` is available in `getBeaconHandler()`.  cf. `class ChannelSearchManager` for an example of using `Timer`.